src/ucp/core/ucp_request.inl

```diff
@@ -570,10 +570,16 @@ ucp_request_recv_data_unpack(ucp_request_t *req, const void *data,
     ucp_dt_generic_t *dt_gen;
     ucs_status_t status;
 
-    ucs_assert(req->status == UCS_OK);
+    ucp_trace_req(req,
+                  "unpack recv_data req_len %zu data_len %zu offset %zu "
+                  "last %s status %s",
+                  req->recv.length, length, offset, last ? "yes" : "no",
+                  ucs_status_string(req->status));
 
-    ucp_trace_req(req, "unpack recv_data req_len %zu data_len %zu offset %zu last: %s",
-                  req->recv.length, length, offset, last ? "yes" : "no");
+    if (ucs_unlikely(req->status != UCS_OK)) {
+        /* in case if status is not OK - just return */
+        return req->status;
```

Comment:
could this fail an assert in `uct_iface_invoke_am`, which expects either `OK` or `INPROGRESS`?

Reply:
evaluation updated

Reply:
this func still can return non `OK` or `INPROGRESS`

Reply:
right, this is correct behavior

Reply:
this cond is already checked by ucp_request_process_recv_data()
i think we can just remove the assert from eager_rcv.c:113

Reply:
done

Reply:
> done

not pushed..

Reply:
it is already there